src/datatrove/pipeline/readers/huggingface.py

```diff
@@ -44,23 +47,53 @@ def __init__(
     ):
         super().__init__(limit, skip, progress, adapter, text_key, id_key, default_metadata)
         self.dataset = dataset
-        self.dataset_options = dataset_options
+        self.dataset_options = dataset_options or {}
         self.batch_size = batch_size
+        self.streaming = streaming
 
     def get_document_from_dict(self, data: dict, source: str, id_in_file: int | str):
         document = super().get_document_from_dict(data, source, id_in_file)
         if document:
             document.metadata.setdefault("dataset", source)
         return document
 
+    def _get_dataset_shard(self, dst, rank: int, world_size: int):
+        from datasets import Dataset, IterableDataset
+        from datasets.distributed import split_dataset_by_node
+
+        if isinstance(dst, Dataset):
+            return dst.shard(world_size, rank, contiguous=True)
+        elif isinstance(dst, IterableDataset) and dst.n_shards > 1:
+            # In case we have more than 1 shard (file), we shard
+            # on shards/file level.
+            ex_iterable = dst._ex_iterable.shard_data_sources(rank, world_size)
+            return IterableDataset(
+                ex_iterable=ex_iterable,
+                info=dst._info.copy(),
+                split=dst._split,
+                formatting=dst._formatting,
+                shuffling=copy.deepcopy(dst._shuffling),
+                distributed=copy.deepcopy(dst._distributed),
+                token_per_repo_id=dst._token_per_repo_id,
+            )
+        else:
+            # If we have just a single shard/file, we shard inter-file
+            return split_dataset_by_node(dst, rank, world_size)
```

Comment:
how is this different that just calling `split_dataset_by_node` ?

Reply:
Yes exactly, we have this same behavior for other readers so I wanted to be consistent.